tree/ntuple/v7/src/RField.cxx

```diff
@@ -189,6 +189,17 @@ ROOT::Experimental::Detail::RFieldBase::~RFieldBase()
 {
 }
 
+std::string ROOT::Experimental::Detail::RFieldBase::GetQualifiedFieldName() const
+{
+   std::string result = GetName();
+   RFieldBase *parent = GetParent();
+   while (parent && !parent->GetName().empty()) {
+      result = parent->GetName() + "." + result;
+      parent = parent->GetParent();
+   }
+   return result;
```

Comment:
Alternative implementation:
```
RFieldBase *parent = GetParent();
if (parent && !parent->GetName().empty())
   return parent->GetQualifiedName() + "." + GetName();
 else
   return GetName();
```

Independently, the code clearly assumes that if `parent->GetName().empty()` then either `parent->GetParent()` is nullptr or does not matter.  Is that really the case?